### PR TITLE
Feature/aa 164 api for data calculation

### PR DESCRIPTION
**Pull request by hasmikharutyunyan**
In this PR I have planned to have differential data calculation. But ef core have not correct support of dictionaries, that is why that functionality will be implemented in other PR after some time.

### CODE REVIEW

File: HappyTravel.Nakijin.Api/Services/Workers/AccommodationPreloader.cs [hunk 1]
```diff
@@ -38,25 +38,39 @@ public class AccommodationPreloader : IAccommodationPreloader
...
+                        .Select(dh=> dh.UpdateTime)
+                        .FirstOrDefaultAsync(cancellationToken);
+
+                   using var supplierAccommodationsPreloadingSpan = tracer.StartActiveSpan(
                         $"{nameof(Preload)} accommodations of {supplier.ToString()}", SpanKind.Internal, currentSpan);
 
                     cancellationToken.ThrowIfCancellationRequested();
-                    await Preload(supplier, modificationDate.Value,
+                    await Preload(supplier, modificationDate,
                         cancellationToken);
```
Comment (kirillta) [inline HappyTravel.Nakijin.Api/Services/Workers/AccommodationPreloader.cs:65]: Formatting

File: HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs [hunk 2]
```diff
@@ -168,6 +177,37 @@ public IActionResult MergeAccommodationsData()
 
             return Accepted();
         }
+        
+        /// <summary>
+        /// Calculates accommodations data
+        /// </summary>
+        /// <returns></returns>
+        [HttpPost("calculate")]
```
Comment (kirillta) [inline HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs:185]: We have `calculate` and `calculation` in routes. That's quite confusing. @mingazhev what do you think?

Comment (mingazhev) [reply HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs:185]: Agree, may we treat this as new "calculation" resource creation?
If not we may form another url, like `calculation/start` or something similar to be similar to another roat

**Review by mingazhev — COMMENTED** [review]
Routes looks different for me, agree with @kirillta

File: HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs [hunk 3]
```diff
@@ -168,6 +177,37 @@ public IActionResult MergeAccommodationsData()
...
+                try
+                {
+                    var accommodationService = scope.ServiceProvider.GetRequiredService<IAccommodationsDataMerger>();
+                    await accommodationService.Calculate(suppliers,_accommodationsDataCalculatorTokenSource.Token);
+                }
+                finally
+                {
+                    scope.Dispose();
+                }
+            }, _accommodationsDataCalculatorTokenSource.Token, TaskCreationOptions.LongRunning, TaskScheduler.Default);
```
Comment (AlexeyShok) [inline HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs:207]: Formatting is very inconvenient. Maybe you should wrap to the next line?

File: HappyTravel.Nakijin.Api/Services/Workers/AccommodationDataMerger.cs [hunk 4]
```diff
@@ -133,6 +168,57 @@ where accommodation.SupplierAccommodationCodes.Values.Contains(ac.SupplierAccomm
...
+                var dbAccommodation = new RichAccommodationDetails();
+                dbAccommodation.Id = ac.Id;
+                dbAccommodation.IsCalculated = true;
+                dbAccommodation.CalculatedAccommodation = calculatedData;
+                dbAccommodation.HasDirectContract = calculatedData.HasDirectContract;
+                dbAccommodation.MappingData =
+                    _multilingualDataHelper.GetAccommodationDataForMapping(calculatedData);
+                dbAccommodation.Modified = DateTime.UtcNow;
+                _context.Accommodations.Attach(dbAccommodation);
+                _context.Entry(dbAccommodation).Property(p => p.CalculatedAccommodation).IsModified = true;
```
Comment (AlexeyShok) [inline HappyTravel.Nakijin.Api/Services/Workers/AccommodationDataMerger.cs:206]: Wouldn't it be better to simplify this and the next 4 lines by using a variable?

File: HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs (hunk 3 above)
Comment (hasmikharutyunyan) [reply HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs:207]: As I remember we decided that as much parameters to place in one raw, than better

**Review by kirillta — APPROVED** [review]

File: HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs (hunk 3 above)
Comment (kirillta) [reply HappyTravel.Nakijin.Api/Controllers/AccommodationsManagementController.cs:207]: As I understand, the initial comment is about putting a new line only after the first comma

**Pull request merged**